CMakeLists.txt

```diff
@@ -52,71 +52,47 @@ ENDIF()
 set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
 "Release" "Debug" "RelWithDebInfo" "MinSizeRel")
 
-project(rascal)
+message(STATUS "Build type is: ${CMAKE_BUILD_TYPE}")
+
+project(rascal CXX)
 
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(BUILD_SHARED_LIBS ON)
-set(RASCAL_PYTHON_MAJOR_VERSION 3)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 include(rascalTools)
 
 ########## MAIN VARIABLE TO CHANGE THE BUILD AND INSTALL OF THE LIBRARY ########
-set(BUILD_BINDINGS "ON" CACHE BOOL
-  "If on, the python bindings will be built.")
-
-set(BUILD_EXAMPLES "ON" CACHE BOOL
-  "If on, the examples will be built.")
-
-set(USER "OFF" CACHE BOOL
-  "Switch installation folder")
+option(BUILD_BINDINGS "Build the python bindings" ON)
+option(BUILD_EXAMPLES "Build the examples" ON)
+option(BUILD_TESTS "Build the unit tests" OFF)
+option(BUILD_DOC "Build documentation" OFF)
+option(USER "Switch installation folder" OFF)
 
-set(INSTALL_PATH "" CACHE STRING
-  "Path to install the libraries")
+set(INSTALL_PATH "" CACHE STRING "Path to install the libraries")
 
-set(ENABLE_DOC "OFF" CACHE BOOL
-  "If on, a target 'dev_doc' (which builds the documentation) is added.")
+SET(TYPE_ARCHITECTURE "native" CACHE STRING "Choose the type of architecture to compile to in release mode. Default is to let the compiler decide")
 
-set(BUILD_TESTS "OFF" CACHE BOOL
-  "If on, builds the unit tests")
+########## COMPILATION FLAGS ##########
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Weffc++ -Wno-non-virtual-dtor")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra")
 
-set(ENABLE_PROFILING "OFF" CACHE BOOL
-  "If on, adds gprof profiling information to some of the examples. \
-  (GNU compilers only)")
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
+    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
+    "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
 
-SET(TYPE_ARCHITECTURE "native" CACHE STRING "Choose the type of architecture to compile to in release mode. Default is to let the compiler decide")
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
```

Comment:
Do we want to enable this flag by default? As mentioned in #121, this cuts down the runtime of tests to 9s from 147s. The build time goes from 11min (at -O0) to 12min (at -Og).